packages/nango-yaml/lib/helpers.ts

```diff
@@ -0,0 +1,192 @@
+
+    const offset = nowMilliseconds % intervalMilliseconds;
+
+    if (isNaN(offset)) {
+        return 0;
+    }
+
+    return offset;
+}
+
+/**
+ * Get Interval
+ * @desc get the interval based on the runs property in the yaml. The offset
+ * should be the amount of time that the interval should be offset by.
+ * If the time is 1536 and the interval is 30m then the next time the sync should run is 1606
+ * and then 1636 etc. The offset should be based on the interval and should never be
+ * greater than the interval
+ */
+export function getInterval(runs: string, date: Date): IntervalResponse | Error {
+    if (runs === 'every half day') {
+        const response: IntervalResponse = { interval: '12h', offset: getOffset('12h', date) };
+        return response;
+    }
+
+    if (runs === 'every half hour') {
+        const response: IntervalResponse = { interval: '30m', offset: getOffset('30m', date) };
+        return response;
+    }
+
+    if (runs === 'every quarter hour') {
+        const response: IntervalResponse = { interval: '15m', offset: getOffset('15m', date) };
+        return response;
+    }
+
+    if (runs === 'every hour') {
+        const response: IntervalResponse = { interval: '1h', offset: getOffset('1h', date) };
+        return response;
+    }
+
+    if (runs === 'every day') {
+        const response: IntervalResponse = { interval: '1d', offset: getOffset('1d', date) };
+        return response;
+    }
+
+    if (runs === 'every month') {
+        const response: IntervalResponse = { interval: '30d', offset: getOffset('30d', date) };
+        return response;
+    }
+
+    if (runs === 'every week') {
+        const response: IntervalResponse = { interval: '1w', offset: getOffset('1w', date) };
+        return response;
+    }
+
+    const interval = runs.replace('every ', '') as StringValue;
+
+    if (!ms(interval)) {
+        return new Error('sync_interval_invalid');
+    }
+
+    if (ms(interval) < ms('5m')) {
+        return new Error('sync_interval_too_short');
+    }
+
+    const offset = getOffset(interval, date);
+    const response: IntervalResponse = { interval, offset };
+
+    return response;
+}
+
+export const JAVASCRIPT_AND_TYPESCRIPT_TYPES = {
+    primitives: ['string', 'number', 'boolean', 'bigint', 'symbol'],
+    builtInObjects: ['Object', 'Array', 'Function', 'Date', 'RegExp', 'Map', 'Set', 'WeakMap', 'WeakSet', 'Promise', 'Symbol', 'Error'],
+    utilityTypes: ['Record', 'Partial', 'Readonly', 'Pick', 'Omit', 'Awaited', 'Required', 'Exclude', 'Extract', 'Uppercase', 'Lowercase']
+};
+const typesLowercase = Object.values(JAVASCRIPT_AND_TYPESCRIPT_TYPES)
+    .flat()
+    .map((v) => v.toLocaleLowerCase());
+const typesWithGenerics = [...JAVASCRIPT_AND_TYPESCRIPT_TYPES.builtInObjects, ...JAVASCRIPT_AND_TYPESCRIPT_TYPES.utilityTypes];
+// Only used externally
+export function isJsOrTsType(type?: string): boolean {
```

Comment:
I moved it before realizing it wasn't what I expected, it will be deleted as soon as I migrate the backend to use the new schema